### ISSUE TITLE
Code on GitHub

### ISSUE THREAD

**Issue by Heyian** [SilverAzide/Gadgets#95]
I'm not sure this is the right place to have this discussion and I couldn't find discussions on this matter. Would it be possible to have the source code on GitHub? It could in a folder called "src' for example.

I've added the DNS provider in Network Meter (under Ext IP) because I switch from my Pi-Hole to QuadDNS when I need to open a support ticket (many chat windows don't open with my Pi-Hole). I've simply copied the .ini file and created my own fork and it works flawlessly. 

But what I'd like to do is simply fork the code on GitHub so I can keep up with updates if there's some. I could also open a PR if that's something that's interesting for you.

Thanks for your time and thanks a lot for this skin, really nice job!

Take care!

**Comment by SilverAzide** [comment]
Hello!

Sure, this is a good a place as any for discussions, unless you want to do it on the Rainmeter forum in the Gadgets thread.  Since we're here already, might as well continue, LOL...

I thought about putting this on GitHub a long time ago, but eventually I decided it wasn't worth the hassle.  By "hassle" I mean that I already have an extensive Subversion repo hosted on my own NAS, which has a lot more stuff in it than just the Gadgets skins.  So to put the Gadgets code on GitHub I'd end up needing to manage two repos.  Plus I keep a lot more stuff in my repo that I wouldn't want on GitHub, like internal documentation, prototypes, artwork, my ActiveNet plugin source, etc.

Since Rainmeter skins are essentially all "source code" anyway, I figure it would be easy enough to for someone to just keep everything in their own repo if they needed one.  For a PR, you could just send me the changed files, and I can merge/cherry-pick the differences into my working branch.  Similarly, when a new version of the Gadgets comes out, you should be able to merge the new files into your code without too much hassle (fingers crossed).

Anyway, it's good to hear you are finding the Gadgets useful for more than just having a bunch of blinky lights and squiggly graphs on your desktop, LOL.  They can be handy tools sometimes...

**Comment by Heyian** [comment]
Makes total sense, I'll add the "Main DNS" field in the config and I'll put a working hide icon next to the DNS value so it doesn't take space for people who wouldn't want it. I haven't implemented this since I knew I wanted it and the Pi-Hole dns value is currently hard coded ;)

I'll post a link in here for you to download the files once this is done.

**Comment by SilverAzide** [comment]
Closing this issue for now; will reopen if OP responds.

**Comment by Heyian** [comment]
I just saw your comment and it made me realize I should at least tell you that a couple of weeks ago I got frustrated at Microsoft for many reasons and I actually jumped ship. I now run an Arch Linux environment 100% of the time and even though I installed it with dual boot, I never even once logged back in my Windows… So I’m kind of done with the gadgets, at least for a while. I can still take some time  to adapt my code and send it to you if you think it would help though.

Thanks again for your great work.
Take care!
